disvae/losses.py

```diff
@@ -190,12 +190,20 @@ class FactorKLoss(BaseLoss):
             arXiv preprint arXiv:1802.05983 (2018).
         """
 
-    def __init__(self, is_color, discriminator, optimizer_d, device, beta=40.):
+    def __init__(self, is_color, device, beta=40.,
+                 disc_kwargs=dict(neg_slope=0.2, latent_dim=10, hidden_units=1000),
+                 optim_kwargs=dict(lr=1e-4, betas=(0.5, 0.9))):
         super().__init__(is_color)
         self.beta = beta
         self.device = device
-        self.discriminator = discriminator
-        self.optimizer_d = optimizer_d
+
+        self.discriminator = Discriminator(neg_slope=disc_kwargs['neg_slope'],
+                                           latent_dim=disc_kwargs['latent_dim'],
```

Comment:
use `Discriminator(**disc_kwargs)`, i.e. just redirect the arguments, such that you can change all arguments of `Discriminator` from the loss (without having to manually redirect as you are doing)

Same for optimizer

Reply:
I tried this and had some issues when passing an empty dictionary. Let me give it another go and see if I can find what the issue was.

Reply:
Not entirely sure what I was doing (clearly something a bit stupid), but is now working.

Reply:
Have changed ...